src/votekit/ballot_generator.py

```diff
@@ -1932,53 +1932,102 @@ def generate_profile(
             return pp
 
 
-class UniformSpatial(BallotGenerator):
+# Not sure where to put this! Poetry complains if I try to put it inside the class
+def euclidean_distance(point1: np.ndarray, point2: np.ndarray) -> float:
+    return float(np.linalg.norm(point1 - point2))
+
+
+class DSpatial(BallotGenerator):
     """
-    Uniform spatial model for ballot generation. Assumes the candidates are uniformly distributed in
-    d dimensional space. Voters are then distributed equivalently, and vote with
-    ballots based on Euclidean distance to the candidates.
+    Spatial model for ballot generation. In some metric space determined
+    by an input distance function, randomly sample each voter's and
+    each candidate's positions from input voter and candidate distributions.
+    Using generate_profile() outputs a ranked profile which is consistent
+    with the sampled positions (respects distances).
 
     Args:
-        candidates (list): List of candidate strings.
-
+        candidates (list[str]): List of candidate strings.
+        voter_dist (Callable[..., np.ndarray]): Distribution to sample a single
+        voter's position from, defaults to uniform distribution.
+        voter_params: (dict[str, float], optional): Parameters to be passed to
+        voter_dist, defaults to uniform(0,1) in 2 dimensions.
```

Comment:
say defaults to None, which creates the unif(0,1) in 2D.

Reply:
Make this `Optional[dict[str, Any]]` in case users have more complicated distributions that need non-float parameters.